substrate-sgx/externalities/src/lib.rs

```diff
@@ -41,72 +49,18 @@ environmental!(ext: SgxExternalities);
 
 pub trait SgxExternalitiesTrait {
 	fn new() -> Self;
-	fn decode(state: Vec<u8>) -> Self;
-	fn encode(self) -> Vec<u8>;
 	fn insert(&mut self, k: Vec<u8>, v: Vec<u8>) -> Option<Vec<u8>>;
 	fn remove(&mut self, k: &[u8]) -> Option<Vec<u8>>;
-	fn get(&mut self, k: &[u8]) -> Option<&Vec<u8>>;
-	fn contains_key(&mut self, k: &[u8]) -> bool;
+	fn get(&self, k: &[u8]) -> Option<&Vec<u8>>;
+	fn contains_key(&self, k: &[u8]) -> bool;
```

Comment:
Nice - with the `mut` requirement, it was probably difficult to use in certain scenarios 👍

Reply:
You are right. At least it demanded some unnecessary `mut`s that propagated surprisingly far.